native/cocos/renderer/gfx-base/GFXBarrier.cpp

```diff
@@ -0,0 +1,269 @@
+     AccessFlags::FRAGMENT_SHADER_READ_UNIFORM_BUFFER},
+
+    {IGNORE_NONE,
+     ACCESS_READ | MEM_DEVICE | PASS_RASTER | RES_TEXTURE | SHADERSTAGE_FRAG | CMN_ROM,
+     AccessFlags::FRAGMENT_SHADER_READ_TEXTURE},
+
+    {IGNORE_NONE,
+     ACCESS_READ | MEM_DEVICE | PASS_RASTER | RES_TEXTURE | SHADERSTAGE_FRAG | CMN_IB_OR_CA | CMN_INDIRECT_OR_INPUT,
+     AccessFlags::FRAGMENT_SHADER_READ_COLOR_INPUT_ATTACHMENT},
+
+    {IGNORE_NONE,
+     ACCESS_READ | MEM_DEVICE | PASS_RASTER | RES_TEXTURE | SHADERSTAGE_FRAG | CMN_VB_OR_DS | CMN_INDIRECT_OR_INPUT,
+     AccessFlags::FRAGMENT_SHADER_READ_DEPTH_STENCIL_INPUT_ATTACHMENT},
+
+    {IGNORE_NONE,
+     ACCESS_READ | MEM_DEVICE | PASS_RASTER | SHADERSTAGE_FRAG,
+     AccessFlags::FRAGMENT_SHADER_READ_OTHER},
+
+    {IGNORE_NONE,
+     ACCESS_READ | MEM_DEVICE | PASS_RASTER | RES_TEXTURE | SHADERSTAGE_FRAG | CMN_IB_OR_CA,
+     AccessFlags::COLOR_ATTACHMENT_READ},
+
+    {IGNORE_NONE,
+     ACCESS_READ | MEM_DEVICE | PASS_RASTER | RES_TEXTURE | SHADERSTAGE_FRAG | CMN_VB_OR_DS,
+     AccessFlags::DEPTH_STENCIL_ATTACHMENT_READ},
+
+    {IGNORE_PASSTYPE & IGNORE_SHADERSTAGE,
+     ACCESS_READ | MEM_DEVICE | RES_BUFFER | CMN_ROM,
+     AccessFlags::COMPUTE_SHADER_READ_UNIFORM_BUFFER},
+
+    {IGNORE_PASSTYPE & IGNORE_SHADERSTAGE,
+     ACCESS_READ | MEM_DEVICE | RES_TEXTURE | CMN_ROM,
+     AccessFlags::COMPUTE_SHADER_READ_TEXTURE},
+
+    {CARE_MEMACCESS | CARE_MEMUSAGE,
+     ACCESS_READ | MEM_DEVICE,
+     AccessFlags::COMPUTE_SHADER_READ_OTHER},
+
+    {CARE_MEMACCESS | CARE_MEMUSAGE | CARE_PASSTYPE,
+     ACCESS_READ | MEM_DEVICE | PASS_COMPUTE,
+     AccessFlags::TRANSFER_READ},
+
+    {CARE_MEMACCESS | CARE_MEMUSAGE,
+     ACCESS_READ | MEM_HOST,
+     AccessFlags::HOST_READ},
+
+    {CARE_MEMACCESS | CARE_MEMUSAGE | CARE_PASSTYPE,
+     ACCESS_READ | MEM_DEVICE | PASS_PRESENT,
+     AccessFlags::PRESENT},
+
+    {IGNORE_RESTYPE & IGNORE_CMNUSAGE,
+     ACCESS_WRITE | MEM_DEVICE | PASS_RASTER | SHADERSTAGE_VERT,
+     AccessFlags::VERTEX_SHADER_WRITE},
+
+    {IGNORE_RESTYPE & IGNORE_CMNUSAGE,
+     ACCESS_WRITE | MEM_DEVICE | PASS_RASTER | SHADERSTAGE_FRAG,
+     AccessFlags::FRAGMENT_SHADER_WRITE},
+
+    {IGNORE_NONE,
+     ACCESS_WRITE | MEM_DEVICE | PASS_RASTER | RES_TEXTURE | SHADERSTAGE_FRAG | CMN_IB_OR_CA,
+     AccessFlags::COLOR_ATTACHMENT_WRITE},
+
+    {IGNORE_NONE,
+     ACCESS_WRITE | MEM_DEVICE | PASS_RASTER | RES_TEXTURE | SHADERSTAGE_FRAG | CMN_VB_OR_DS,
+     AccessFlags::DEPTH_STENCIL_ATTACHMENT_WRITE},
+
+    {CARE_MEMACCESS | CARE_MEMUSAGE,
+     ACCESS_WRITE | MEM_DEVICE,
+     AccessFlags::COMPUTE_SHADER_WRITE},
+
+    {CARE_MEMACCESS | CARE_MEMUSAGE | CARE_PASSTYPE,
+     ACCESS_WRITE | MEM_DEVICE | PASS_COPY,
+     AccessFlags::TRANSFER_WRITE},
+
+    {CARE_MEMACCESS | CARE_MEMUSAGE,
+     ACCESS_WRITE | MEM_HOST,
+     AccessFlags::HOST_WRITE},
+}};
+
+} // namespace
+
+AccessFlags getAccessFlags(BufferUsage usage, MemoryUsage memUsage,
```

Comment:
These functions are from framegraph execution. They will be reused in other places.